module4-authentication-and-security/securing-backend-app/README.md

```diff
@@ -0,0 +1,79 @@
+INNER JOIN `items`
+ON `items`.`list_id` = `lists`.`id`
+WHERE `lists`.`user_email` = "test@domain.com"
+```
+
+Thus, when a user comes to your app, you can ask them first about their email address, and then query the database to get thier lists. But that is also insufficient. A malicious user can peep on other users wishlists if he/she knew thier emails, or he/she check different email combinations to see if it lands a hit, aka, **Brute Force**.
+
+To make this a bit harder to brute force, you can create a token system. Tokens are random strings of charcters, usually wih arbitrary length, that is intended to be hard to brute force (guess). A token can look something like this: `1FsahNjuio31OHsOallsn1gHH7`
+
+You can change the column `user_email` to `token` and everytime a user comes in, you can ask them for the token, or give them the opportunity to generate a token. If they are new users, they can generate a new token, and save it, then start creating thier lists. When they visit again, they enter that token to get thier lists back.
+
+Bingo! You've just created your first, very acceptable authentication layer! You've secured your app and **slimmed the chances** of brute forcing user data! Many apps today use this pattern to secure data, however it is not always acceptable. That's because, if a user lost thier token, it is next to impossible (especially if the token was generated by a cryptograpic random generators with reasonable length) to to gain their access back to their lists using today's technologies.
+
+## Why we need passwords
+
+The token system developed above is very robust, but it isn't always acceptable security measure (at least not alone). That's because, if a user lost thier token, it is next to impossible (especially if the token was generated by a cryptograpic random generators with reasonable length) to to gain their access back to their lists using today's technologies.
+
+So, maybe instead of a long random token, use a password? We can ask the user when they land on our page for a pass code (aka password). And we check the database, if we can find it a list where `password = USER_SUBMITTED_PASS_CODE` then display them. If not, that means they are a new user.
+
+Indeed this works, but we are back to the same problem with using emails. Users tend to use easy to remember words, dates, phone numbers, and the like. It would be relatively harder than bruteforcing emails, but nonetheless, still easy to guess, and very easy for computers to break by checking every possible combination of words, dates, numbers, and even relatively harder passwords using a technology called **rainbow tables**.
+
+**However**, if you do more validation on yourside to limit the possibilities of such a risk, then passwords can work. In the end, almost all web apps use passwords for authentication. Below are the highlights of best practices to securing with passwords:
+
+- Use 3rd party identity providers: Sign in with Google
```

Comment:
there are too many bullet points here -- hurts readability to just have a long list. each of these should probably be paragraphs, perhaps organized by sections.

also i think that each of these has to be expanded on, to most students these statements will be meaningless